sphinx/templates/imgmath/preview.tex_t

```diff
@@ -9,7 +9,7 @@
 \pagestyle{empty}
 <%= preamble %>
 
-\usepackage[active<%= tightpage %>]{preview}
+\usepackage[active<%= tightpage %><%= dvips %>]{preview}
```

Comment:
It could be ``\usepackage[active,dvips<%=tightpage %>]{preview}`` as argued above.

Reply:
I have made quite some testing, and `dvips` option seems to have nil impact on the `dvipng` path.  So we can have it here always because:
- it does not (in my testing) change anything when image format is `png`.
- it is already implied by `tightpage` option with standard `latex`,
- with `xelatex`, it is needed in addition to `tightpage` else `dvisvgm` can not report the found depth

The `tightpage` is needed when format is `svg` else `dvisvgm` can not report an image depth.  With standard `latex`(not `pdflatex` or `xelatex`)  the `tightpage` option of `preview` automatically activate the `dvips` option, and `dvisvgm` can extra depth information.

So we could almost use always `dvips,tightpage` except that in my testing `tightpage` with `png` image format causes slight modification compared to not using it.  The depth is different (it seems by 1pt) and the picture width also it seems, slightly.

Thus I propose with keep `tightpage` conditional on image format being `svg` but we can use `dvips` always: does not change anything to the `png` output from `dvipng`, is already tacitly implied if image format is `svg` and using `latex`, is needed in this PR if using `xelatex/tectonic`.

Reply:
I have already addressed these issues above. Here I summarize the cases:

# Cases

Note: `xelatex` covers the case of `tectonic` as well (with only one difference: the `--interaction` parameter).

## PNG

### latex

+ `tightpage` is not used
+ `dvips` option has no impact on PNG files (empirically verified)
+ `dvips` option results in a different `stdout` from `dvipng` (with the standard options) and requires a slight tweaking of `depth_re`

### xelatex

+ cannot be used
+ `xelatex` outputs XDV file (i.e., DVI version 7) and `dvipng` cannot use it

## SVG

### latex

+ `tightpage` is used
+ `dvips` has no impact on SVG files (empirically verified - I didn't understand why it is implied by `tightpage` in the code snippet from `prtightpage.def`: my ability to read such latex code is limited)

### xelatex

+ `tightpage` is used
+ `dvips` is used (otherwise depth information is not displayed by `dvisvgm`)

Reply:
Your summary appears correct to me; and as we have discusses `dvips` is automatically triggered by `preview` if it has received `tightpage` option and is used with `latex`.  So using `dvips,tightpage` for `'svg'` is no change at all to latex.  It would be a change to `pdflatex` but `imgmath` has not so far supported `'pdf'` as intermediate image format.  And `dvips,tightpage` is required if we cant `xelatex` to work here.

As per the `luatex`.  What happens is that I considered so far main reason for perhaps not using `latex` but one of `xelatex/lualatex` is for matters of math fonts.  Thanks to you, I realized today that one can actually use already `dvilualatex` as a replacement of `latex` for the `imgmath_latex` command.  It was not the case some years ago if one wanted `'svg'` for image format.  It works today with current `dvisvgm`.  So the user can, using `dvilualatex` add a custom usage of OpenMath fonts, which `latex` can not use, only `lualatex` or `xelatex`.

So here I saw only need for perhaps some addition to document, rather than adding classes.  The existing code seemed generic enough (apart from not handling `xdv` but this is a bit normal as `dvipng` can not be used with it).  Also I am a bit prejudiced against XeTeX in matters of TeX math: it is known to have a number of bugs in the math rendering which have not been fixed for years.  Now if in future the `tectonic` project achives a rewrite of XeTeX in Rust perhaps it will become attractive for speed...

The priority-one thing would be to find a way to execute latex or xelatex or dvilualatex only once, on one file containing all math snippets.  Both of `dvipng` and `dvisvgm` can handle multiple pages dvi.  The speed gain would be tremendous. I did a test with only 100 equations and it takes (on my slow old laptop, divide by 5 to 10 on current ones) about 50 seconds.  [Caching the LaTeX preamble](https://github.com/jfbu/sphinx/tree/imgmath_cache_latexfmt) I could achieve about 10% gain.  And this is impossible with `lualatex/xelatex` if OpenType fonts are used. This is not enough gain.

I have some ideas about how to proceed but now I will not have the time for it in next few weeks probably (I said that here already close to a month ago, but now this is for real...)